common.py

```diff
@@ -61,10 +61,9 @@ def load_from_cache(layer):
 
 def save_to_cache(image):
     '''Given an image object, check against the cache to see if a layer id
-    exists. If no then save the layer and package list to the cache'''
-    existing_layers = cache.get_layers()
+    exists. If no then save the package list to the cache'''
     for layer in image.layers:
-        if layer.fs_hash not in existing_layers and layer.packages:
+        if layer.fs_hash not in layer.packages:
```

Comment:
We're only checking if the layer object has a non-empty package list. You can do that in python by just checking to see if layer.packages (`packages` is a member layer and it is a python list) exists. Like this: `if layer.packages:`